src/runtime_src/core/common/system.h

```diff
@@ -106,6 +106,14 @@ class system
   virtual std::shared_ptr<device>
   get_mgmtpf_device(device::id_type id) const = 0;
 
+  /**
+   */
+  virtual std::string
+  get_hostname() const
+  {
+    return std::string("hostname");
```

Comment:
I was not sure as to what should be the default action for the emulation system files. Let me know what you all think might be better please!

Reply:
I am still looking at this rather complicated implementation of hostname().  Why does host name need to be determined at the shim level?   Does the function have to be virtual and redefined the same way in all the core shim libraries?

Reply:
You have implemented it the same on windows and linux. There is no OS specific handling.   The GNUC conditional is apparently not needed, and taking this simple function all the way down the shim chain is just too much even compared to conditional if that was necessary.

I am fine moving the function to system.h and system.cpp, but leave it there and don't virtualize it.

Reply:
sounds good. I will move it into system.h/.cpp!